pkg/compiler/analysis.go

```diff
@@ -40,6 +40,22 @@ func (c *codegen) getIdentName(pkg string, name string) string {
 // and returns number of variables initialized and
 // true if any init functions were encountered.
 func (c *codegen) traverseGlobals() (int, bool) {
+	var hasDefer bool
+	c.ForEachFile(func(f *ast.File, _ *types.Package) {
```

Comment:
I fear we might be doing way too many passes over the code.

Reply:
`countGlobals` is will become too messy with this. Added to iteration searching for `init` functions.